action.yml

```diff
@@ -1,60 +1,188 @@
+          pip install -e ./tutor
+        else
+          pip install "tutor$INPUT_TUTOR_VERSION"
+        fi
+
+        tutor config save --set LMS_HOST=$LMS_HOST --set CMS_HOST=$CMS_HOST
+        tutor local launch -I
+      shell: bash
+      env:
+        INPUT_TUTOR_VERSION: ${{ inputs.tutor_version }}
+
+    - name: Configure Caddyfile and Open edX settings
+      run: |
+        source .tutor_venv/bin/activate
+
+        mkdir -p plugins
+        cp $GITHUB_ACTION_PATH/patches.yml plugins/
+        tutor plugins enable patches
+      shell: bash
+
+    - name: Add mount for Open edX plugin
+      run: |
+        source .tutor_venv/bin/activate
+
+        tutor mounts add "lms:$GITHUB_WORKSPACE/${{ inputs.app_name }}:/openedx/${{ inputs.app_name }}"
+        tutor mounts add "cms:$GITHUB_WORKSPACE/${{ inputs.app_name }}:/openedx/${{ inputs.app_name }}"
+      shell: bash
+
+    - name: Restart Tutor services to apply mounts
+      run: |
+        source .tutor_venv/bin/activate
+
+        tutor local stop lms cms
+        tutor local start -d lms cms
+      shell: bash
+
+    - name: Install Open edX plugin as an editable package
+      run: |
+        source .tutor_venv/bin/activate
+
+        tutor local exec lms pip install -e /openedx/${{ inputs.app_name }}/
+        tutor local exec cms pip install -e /openedx/${{ inputs.app_name }}/
+      shell: bash
+
+    - name: Install extra requirements
+      run: |
+        source .tutor_venv/bin/activate
+
+        tutor local exec lms pip install ${{ inputs.openedx_extra_pip_requirements }}
+        tutor local exec cms pip install ${{ inputs.openedx_extra_pip_requirements }}
+      shell: bash
+
+    - name: Run migrations and restart services
+      run: |
+        source .tutor_venv/bin/activate
+
+        tutor local exec lms python manage.py lms migrate
+        tutor local exec cms python manage.py cms migrate
+        tutor local restart
+      shell: bash
+
+    - name: Import Demo course
+      run: |
+        source .tutor_venv/bin/activate
+
+        tutor local do importdemocourse
+      shell: bash
+
+    - name: Test Open edX imports in plugin
+      if: ${{ inputs.openedx_imports_test_file_path && inputs.openedx_imports_test_function_name }}
+      run: |
+        source .tutor_venv/bin/activate
+
+        cp $GITHUB_ACTION_PATH/run_openedx_imports_test.py $GITHUB_WORKSPACE/${{ inputs.app_name }}/
+        echo "Running test for Open edX imports in plugin"
+
+        tutor local exec --env PYTHONPATH=/openedx/${{ inputs.app_name }} \
+                        --env DJANGO_SETTINGS_MODULE=lms.envs.production \
+                        lms python /openedx/${{ inputs.app_name }}/run_openedx_imports_test.py \
+                        --test-file-path /openedx/${{ inputs.app_name }}/${{ inputs.openedx_imports_test_file_path }} \
+                        --test-function-name '${{ inputs.openedx_imports_test_function_name }}'
```

Comment:
Should we use environment variables for these long paths?

```
env:
     OPENEDX_IMPORTS_TEST_PATH: /openedx/${{ inputs.app_name }}/run_openedx_imports_test.py
     TEST_FILE_PATH: /openedx/${{ inputs.app_name }}/${{ inputs.openedx_imports_test_file_path }}
```

Reply:
Much better. Thanks